include/mbedtls/ccm.h

```diff
@@ -40,6 +42,106 @@ typedef struct {
 }
 mbedtls_ccm_context;
 
+#else /* MBEDTLS_CCM_ALT */
+#include "ccm_alt.h"
+
+typedef struct {
+    union
+    {
+        mbedtls_ccm_alt_context_t ccm_alt_ctx;    /*!< opaque context structure to be defined in ccm_alt.h */
+        mbedtls_cipher_context_t cipher_ctx;
+    }u_ctx;
+    unsigned int    is_alternative;
```

Comment:
Usual practice is to put the distinguisher at the beginning. This is very useful when debugging, and preserves binary compatibility when adding or changing options. Why is the distinguisher at the end here?